### PR TITLE
feat: (IAC-578) SSL Configuration Support for Postgres

### PR DESCRIPTION
**Pull request by jarpat**
## Changes
1. Introduces the `server_ssl_key_file` and `server_ssl_cert_file` variables that will allow the user to set their Postgres cert and key file if they choose to turn on SSL with `server_ssl`
2. If you set server_ssl to on, and you do not define either `server_ssl_cert_file` or `server_ssl_cert_file` the system's default SSL certificate and key will be used instead. By default, on Ubuntu systems we create a copy of those files and name them `ssl-cert-sas.pem` and `ssl-cert-sas.key` and update the `postgres.conf`. Ansible will take care of copying those certs back into your local workspace directory so that you can use them with viya4-deployment.

Note: If you are planning on using the viya4-deployment repository to perform a Viya deployment where with full-stakck TLS, ensure in the viya4-deployment ansible-vars.yaml the `V4_CFG_TLS_TRUSTED_CA_CERTS` variable points to a directory that contains the `server_ssl_cert_file`.

## Tests
| Scenario | K8s Version | Postgres Server SSL  | Postgres Cert & Key  | Cadence   | V4_CFG_TLS_GENERATOR | V4_CFG_TLS_MODE | V4_CFG_TLS_CERT & V4_CFG_TLS_KEY | V4_CFG_TLS_TRUSTED_CA_CERTS | Deployment Stabilized |
|----------|-------------|----------------------|----------------------|-----------|----------------------|-----------------|----------------------------------|-----------------------------|-----------------------|
| 1        | 1.22.10     | on                   | not supplied         | Fast:2020 | openssl              | full-stack      | set                              | set & includes postgres pem | yes                   |
| 2        | 1.22.10     | on                   | not supplied         | Fast:2020 | cert-manager         | full-stack      | set                              | set & includes postgres pem | yes                   |
| 3        | 1.22.10     | on                   | supplied             | Fast:2020 | openssl              | full-stack      | set                              | set & includes postgres pem | yes                   |
| 4        | 1.22.10     | on                   | supplied             | Fast:2020 | cert-manager         | full-stack      | set                              | set & includes postgres pem | yes                   |

### CODE REVIEW

**Comment by jarpat** [comment]
This is ready for review, but probably going in the release after the currently planned one.

**Review by riragh — APPROVED** [review]
LGTM

File: templates/ansible/inventory.tmpl [hunk 1]
```diff
@@ -73,11 +73,13 @@ nfs_server
...
 [${prefix}_${server_name}_pgsql:vars]
-postgres_server_name=${server_name}
-postgres_server_version=${server_data.server_version}
-postgres_server_ssl=${server_data.server_ssl}  # NOTE: Values - [on,off]
-postgres_administrator_login=${server_data.administrator_login} # NOTE: Do not change this value at this time
-postgres_administrator_password=${server_data.administrator_password}
+postgres_server_name="${server_name}"
+postgres_server_version="${server_data.server_version}"
+postgres_server_ssl="${server_data.server_ssl}"  # NOTE: Values - [on,off]
+postgres_administrator_login="${server_data.administrator_login}" # NOTE: Do not change this value at this time
```
Comment (thpang) [inline templates/ansible/inventory.tmpl:79]: The # NOTE items have been removed from the template. The updates are currently in staging.

File: roles/systems/postgres/defaults/main.yaml [hunk 2]
```diff
@@ -0,0 +1,2 @@
+postgres_server_ssl_cert_file: ""
+postgres_server_ssl_key_file: ""
```
Comment (thpang) [inline roles/systems/postgres/defaults/main.yaml:2]: Not a fan of no line endings in files. This will pass, but in the future validate.

**Review by thpang — CHANGES_REQUESTED** [review]
Looks good but need a 15 minute meeting to talk through steps.

File: roles/systems/postgres/defaults/main.yaml (hunk 2 above)
Comment (jarpat) [reply roles/systems/postgres/defaults/main.yaml]: Fixed this

File: templates/ansible/inventory.tmpl (hunk 1 above)
Comment (jarpat) [reply templates/ansible/inventory.tmpl]: The notes have been removed, merged staging into this branch to pick up on those changes.

**Review by thpang — APPROVED** [review]
LGTM.

**Pull request merged**